Fix caching issue whereby changes to CSS are not showing in the browser

jsmcgd opened the pull request:
New changes to the stylesheet were not being reflected in all browsers.
A hard refresh of the browser fixed the issue. This suggested that
cache-busting was not enabled for this site. A quick inspection proved
this to be true.

As we are using webpack.mix.js we can take advantage of its
cache-busting functionality
(https://laravel.com/docs/6.x/mix#versioning-and-cache-busting).

Using a custom function `mix_asset` in
`web/app/themes/taxcentreofexcellence/lib/setup.php` we can take
advantage of the `mix-manifest.json` file that is automatically
generated when the `dist` assets are built.

The `mix-asset` function was adapted from the `mix-asset` function in
the wp-ppj project
(https://github.com/ministryofjustice/wp-ppj/blob/master/web/app/themes/ppj/functions.php).

Other sites should be checked to see if they have the same issue.

jsmcgd committed:
Fix caching issue whereby changes to CSS are not showing in the browser

New changes to the stylesheet were not being reflected in all browsers.
A hard refresh of the browser fixed the issue. This suggested that
cache-busting was not enabled for this site. A quick inspection proved
this to be true.

As we are using webpack.mix.js we can take advantage of its
cache-busting functionality
(https://laravel.com/docs/6.x/mix#versioning-and-cache-busting).

Using a custom function `mix_asset` in
`web/app/themes/taxcentreofexcellence/lib/setup.php` we can take
advantage of the `mix-manifest.json` file that is automatically
generated when the `dist` assets are built.

The `mix-asset` function was adapted from the `mix-asset` function in
the wp-ppj project
(https://github.com/ministryofjustice/wp-ppj/blob/master/web/app/themes/ppj/functions.php).

Other sites should be checked to see if they have the same issue.

RobLoweMOJ reviewed (approved):
Looks Good. Please test on AWS and if there are no problems we need to look at using this on other stylesheets included.

jsmcgd committed:
Enqueue all scripts using mix_asset function

All assets (css and js) are now being moved to the `dist` directory so
that they can be enqueued using the mix_asset function. This allows us
to remove any references to the Sage function `Assets\asset_path`.
